### PR TITLE
Bugfix FXIOS-4993 [v108] Tab tray unusuable in split mode on iPad

### PR DESCRIPTION
**Pull request by thatswinnie**
[Jira](https://mozilla-hub.atlassian.net/browse/FXIOS-4993)
#12013

Simulator Screen Shot - iPad Air (5th generation) - 2022-10-24 at 11 45 13
Simulator Screen Shot - iPad Air (5th generation) - 2022-10-24 at 11 45 07

### CODE REVIEW

**Comment by thatswinnie** [comment]
Build passed
Screen Shot 2022-10-24 at 14 30 13

File: Client/Application/AccessibilityIdentifiers.swift [hunk 1]
```diff
@@ -101,6 +101,11 @@ public struct AccessibilityIdentifiers {
         static let syncedTabs = "Synced Tabs"
         static let inactiveTabHeader = "InactiveTabs.header"
         static let inactiveTabDeleteButton = "InactiveTabs.deleteButton"
+        static let closeAllTabsButton = "closeAllTabsButtonTabTray"
+        static let newTabButton = "newTabButtonTabTray"
+        static let doneButton = "doneButtonTabTray"
+        static let syncTabsButton = "syncTabsButtonTabTray"
+        static let navBarSegmentedControl = "navBarTabTray"
```
Comment (lmarceau) [inline Client/Application/AccessibilityIdentifiers.swift:108]: Just to note, if we add new AccessibilityIdentifers we should also replace them in tests classes where they are used. Just to make sure we're not in a weird state where our prod code use the identifiers but not the test code.

File: Client/Frontend/Browser/Tabs/TabTrayViewController.swift [hunk 2]
```diff
@@ -214,65 +221,59 @@ class TabTrayViewController: UIViewController, Themeable {
...
     }
 
-    fileprivate func iPadViewSetup() {
-        navigationItem.leftBarButtonItem = deleteButton
-        navigationItem.titleView = navigationMenu
-        navigationItem.rightBarButtonItems = [doneButton, fixedSpace, newTabButton]
-
-        navigationItem.titleView?.snp.makeConstraints { make in
-            make.width.equalTo(343)
+    fileprivate func updateTitle() {
```
Comment (yoanarios) [inline Client/Frontend/Browser/Tabs/TabTrayViewController.swift:258]:
```suggestion
    private func updateTitle() {
```
nit: replace fileprivate

File: Client/Frontend/Browser/Tabs/TabTrayViewController.swift [hunk 3]
```diff
@@ -339,16 +340,16 @@ class TabTrayViewController: UIViewController, Themeable {
 
     private func updateToolbarItems(forSyncTabs showSyncItems: Bool = false) {
         if shouldUseiPadSetup() {
-            if navigationMenu.selectedSegmentIndex == TabTrayViewModel.Segment.syncedTabs.rawValue {
-                navigationItem.rightBarButtonItems = (showSyncItems ? [doneButton, fixedSpace, syncTabButton] : [doneButton])
+            if segmentedControlIphone.selectedSegmentIndex == TabTrayViewModel.Segment.syncedTabs.rawValue {
```
Comment (yoanarios) [inline Client/Frontend/Browser/Tabs/TabTrayViewController.swift:343]: This func is hard to read there are a few `if cases` and it looks like Sync Tab has custom configuration I think we could have a private func for sync that returns the `rightBarButtonItems` to help with readability and we could unit test

File: Client/Frontend/Browser/Tabs/TabTrayViewController.swift [hunk 4]
```diff
@@ -362,23 +363,105 @@ class TabTrayViewController: UIViewController, Themeable {
...
+        super.traitCollectionDidChange(previousTraitCollection)
+        applyTheme()
+
+        if previousTraitCollection?.horizontalSizeClass != traitCollection.horizontalSizeClass
+            || previousTraitCollection?.verticalSizeClass != traitCollection.verticalSizeClass {
+            updateLayout()
+        }
+    }
+
+    /// On iPhone, we call updateLayout when the trait collection has changed, to ensure calculation
```
Comment (yoanarios) [inline Client/Frontend/Browser/Tabs/TabTrayViewController.swift:408]: 😍

**Review by yoanarios — APPROVED** [review]

**Review by lmarceau — COMMENTED** [review]
Is there a reason why the tab tray title was removed on iPhone with this PR? Is this expected? 
Screen Shot 2022-10-27 at 10 05 45 AM

**Review by lmarceau — APPROVED** [review]

**Review by yoanarios — APPROVED** [review]

**Pull request merged**